docs/source/pages/cli_usage/command_analyze.rst

```diff
@@ -68,12 +69,20 @@ Options
 
 .. option:: --skip-deps
 
-    Skip automatic dependency analysis.
+    DEPRECATED. Dependency resolution is off by default. This flag does nothing and will be removed in the next release.
+
+.. option:: --deps-depth DEPS_DEPTH
+
+    The depth of the dependency resolution. 0: disable, 1: direct dependencies, inf: all transitive dependencies. (Default: 0)
 
 .. option:: -g TEMPLATE_PATH, --template-path TEMPLATE_PATH
 
     The path to the Jinja2 html template (please make sure to use .html or .j2 extensions).
 
+.. option::  --python-venv PYTHON_VENV
+
+    The path to the Python virtual environment of the target software component. If this is set, dependency resolution must be enabled with '--deps-depth'.
```

Comment:
The Python virtual environment can be used for purposes other than dependency resolution, e.g., computing the hash of the package or analyzing its content. So, it should not be necessary to pass `--deps-depth` if the virtual environment path is provided.

Reply:
Do you think we should do this to the SBOM path as well (i.e. the user can both provide a path to an SBOM using `--sbom-path` while disabling dependency resolution with `--deps-depth=0`) ?

Reply:
I have modified the behavior to let SBOM and Python venv provided without having to enable dependency analysis - https://github.com/oracle/macaron/pull/840/commits/7d2ab75006b729bed17633a7740a2b7b8aa7fcd2